skins/laika/src/components/pages/membership_form/Address.vue

```diff
@@ -0,0 +1,150 @@
+import Postal from '@/components/pages/donation_form/Postal.vue';
+import DateOfBirth from '@/components/pages/donation_form/DateOfBirth.vue';
+import ReceiptOptOut from '@/components/pages/donation_form/ReceiptOptOut.vue';
+import Email from '@/components/pages/donation_form/Email.vue';
+import NewsletterOptIn from '@/components/pages/donation_form/NewsletterOptIn.vue';
+import { AddressValidity, FormData, ValidationResult } from '@/view_models/Address';
+import { Validity } from '@/view_models/Validity';
+import { NS_MEMBERSHIP_ADDRESS } from '@/store/namespaces';
+import { setAddressField, validateAddress } from '@/store/address/actionTypes';
+import { action } from '@/store/util';
+
+export default Vue.extend( {
+	name: 'Address',
+	components: {
+		Name,
+		Postal,
+		DateOfBirth,
+		ReceiptOptOut,
+		AddressType,
+		Email,
+		NewsletterOptIn,
+	},
+	data: function (): { formData: FormData } {
+		return {
+			formData: {
+				salutation: {
+					name: 'salutation',
+					value: '',
+					pattern: '^(Herr|Frau)$',
+					optionalField: false,
+				},
+				title: {
+					name: 'title',
+					value: '',
+					pattern: '',
+					optionalField: true,
+				},
+				companyName: {
+					name: 'companyName',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				firstName: {
+					name: 'firstName',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				lastName: {
+					name: 'lastName',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				street: {
+					name: 'street',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				city: {
+					name: 'city',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				postcode: {
+					name: 'postcode',
+					value: '',
+					pattern: '^[0-9]{4,5}$',
+					optionalField: false,
+				},
+				country: {
+					name: 'country',
+					value: 'DE',
+					pattern: '',
+					optionalField: false,
+				},
+			},
+		};
+	},
+	props: {
+		validateAddressUrl: String,
+		countries: Array as () => Array<String>,
+	},
+	computed: {
+		fieldErrors: {
+			get: function (): AddressValidity {
+				return Object.keys( this.formData ).reduce( ( validity: AddressValidity, fieldName: string ) => {
+					if ( !this.formData[ fieldName ].optionalField ) {
+						validity[ fieldName ] = this.$store.state.membership_address.validity[ fieldName ] === Validity.INVALID;
+					}
+					return validity;
+				}, ( {} as AddressValidity ) );
+			},
+		},
+		...mapGetters( NS_MEMBERSHIP_ADDRESS, [
+			'addressType',
+		] ),
+	},
+	methods: {
+		validateForm(): Promise<ValidationResult> {
+			return this.$store.dispatch( action( NS_MEMBERSHIP_ADDRESS, validateAddress ), this.$props.validateAddressUrl );
+		},
+		onFieldChange( fieldName: string ): void {
+			this.$store.dispatch( action( NS_MEMBERSHIP_ADDRESS, setAddressField ), this.formData[ fieldName ] );
+		},
+	},
+} );
+</script>
+<style lang="scss" scoped>
```

Comment:
I think we should open a discussion about scoped SCSS some time. This creates CSS selectors with data attributes which do not perform that well and it's something we should avoid unless really necessary in my opinion. The SCSS rules here may be better placed in a more general location for better reusability and performance.

Reply:
Yeah this should probably be extracted, it's the same rules for the buttons on the donation form.